service/history/shard/context_impl.go

```diff
@@ -1430,6 +1485,26 @@ func (s *ContextImpl) rUnlock() {
 	s.rwLock.RUnlock()
 }
 
+func (s *ContextImpl) ioSemaphoreAcquire(
+	ctx context.Context,
+) (retErr error) {
+	handler := s.metricsHandler.WithTags(metrics.OperationTag(metrics.ShardInfoScope))
+	handler.Counter(metrics.SemaphoreRequests.GetMetricName()).Record(1)
```

Comment:
do you need this? could use count of latency metric?

Reply:
I'd prefer not to rely on the internal implementation of the metrics backend. A different metrics backend may not have the count in latency metric.